pkg/services/alerting/conditions/reducer.go

```diff
@@ -141,8 +145,7 @@ func calculateDiff(series *tsdb.TimeSeries, allNull bool, value float64, fn func
 		for i := 0; i < len(points); i++ {
 			if points[i][0].Valid {
 				allNull = false
-				val := fn(first, points[i][0].Float64)
-				value = math.Abs(val)
+				value = fn(first, points[i][0].Float64)
```

Comment:
I agree that this makes sense but this will be a breaking change.

Reply:
Perhaps something along these lines would make more sense as it doesn't break any existing functionality:

Add additional boolean operator to calculcateDiff function to note if its signed or not
```go
func calculateDiff(series *tsdb.TimeSeries, allNull bool, value float64, fn func(float64, float64) float64, signed bool) (bool, float64) {
```

```go
        case "diff":
		allNull, value = calculateDiff(series, allNull, value, diff, false)
	case "diff_signed":
		allNull, value = calculateDiff(series, allNull, value, diff, true)
	case "percent_diff":
		allNull, value = calculateDiff(series, allNull, value, percentDiff, false)
	case "percent_diff_signed":
		allNull, value = calculateDiff(series, allNull, value, percentDiff, true)
```


```suggestion
				val := fn(first, points[i][0].Float64)
				if !signed {
					value = math.Abs(val)
				} else {
					value = val
				}
```

Reply:
@bergquist sorry I couldn't understand how this change can break any existing functionality. Here I just moved the Abs function inside "var" and assigning the output result directly to "value" variable.

Can you pls help me understand the breaking point with some example?

Reply:
Before your change, `diff` and `percent_diff` was always using `math.Abs()`. 
With this change `diff` and `perfect_diff` with return a signed value. 

With @Spengreb suggestion it `diff` will keep returning an absolute number. So all active alert rule won't change behavior. 

I think `diff_abs` is a more common name but if we use `diff_signed` we avoid breaking changes. Perhaps we should have `diff_abs` and `diff_signed` to make it more clear what the diff returns.

Reply:
So something like this?

```go
       case "diff_abs":
		allNull, value = calculateDiff(series, allNull, value, diff, false)
	case "diff_signed":
		allNull, value = calculateDiff(series, allNull, value, diff, true)
	case "percent_diff_abs":
		allNull, value = calculateDiff(series, allNull, value, percentDiff, false)
	case "percent_diff_signed":
		allNull, value = calculateDiff(series, allNull, value, percentDiff, true)
```

Reply:
@bergquist Initially `diff` was not using `math.Abs()` but after a recent commit to [Extract common code for diff calculation](https://github.com/grafana/grafana/commit/28eaac3a9c7082e8c496005c1cb66b4b70a4f82f#diff-b5991af7e3497d6b806a6cb4659fa174) this breaking change was introduced ....

We were using `diff` with signed value in our alerting but after that recent commit we started facing issues....This PR is an attempt to revert the old breaking change along with adding new aggregators like `diff_abs` and `percent_diff_abs`

moreover using `diff_abs` and `diff_signed` will also be a breaking change as we are already using `diff` aggregator in our alerting. However if you suggest, I can rename `diff` and `percent_diff` to `diff_signed` and `percent_diff_signed` respectively

@spengreb As I had earlier mentioned, your logic of deciding `abs` or `signed` won't be true for some corner cases. feel free to run the test cases written by me to uncover the failing cases.

Reply:
@vikkyomkar Thank you for finding that commit. 

As much as I dislike breaking changes I think we should introduce this change and break the behavior we introduced with 28eaac3a9c7082e8c496005c1cb66b4b70a4f82f .

Could you please merge the suggestions for the tests?

Reply:
@bergquist test case values are updated